### PR TITLE
Orbis 1.18

### LINKED ISSUE #325: Cooler Caves replaces nether portal with water
When entering a new part of the world, from the nether to the overworld, the nether portal gets replaced by the cooler caves module with water in snowy biomes.

1 - Build a nether portal in the nether where a snowy biome would be in the overworld
2 - Go through the portal
3 - In the overworld, the portal should be replaced with water once Orbis and Cooler Caves ahs processed the chunk

**Expected behavior**
Nether portal not replaced by water

**Fix**
Not sure

**Environment**
 - Minecraft version: 1.14.4
 - Installed Datapacks: Orbis, Cooler Caves, & some others

**Additional context**
I used the seed 113318802, I think the world was generated in 1.13, but I'm now playing it on 1.14.4
The coords of the portal I built in the nether are 424 ~ 478

### PR DESCRIPTION
**Pull request by misode**
## Changes
* [x] **Orbis Base**
  * [x] Instead of filling Y=0 with barriers, use markers to store the status of 3x3 chunks "metachunks"
  * [x] Handles all (needs testing) edge cases of slow chunk loading
  * [x] Simplifies biome detection. No more accounting for rivers, they've become too big. Expansions need to add biome predicates themselves.
  * [x] Upgrade old Orbis chunks
* [x] **Cooler Caves**
  * [x] Replace gravel with powder snow
  * [x] Fixes #325 by replacing lava with obsidian
* [x] **Dangerous Dungeons** and **Tower Structures**
  * [x] Increase height spawning range
  * [x] Rewrite randomization
  * [ ] Update height dependent loot

### CODE REVIEW

File: gm4_orbis/data/gm4_orbis/functions/chunk/upgrade.mcfunction [hunk 1]
```diff
@@ -0,0 +1,4 @@
+fill ~ -63 ~ ~15 0 ~15 minecraft:deepslate
+clone ~ 1 ~ ~15 4 ~15 ~ -64 ~ filtered minecraft:bedrock
+fill ~ 1 ~ ~15 4 ~15 minecraft:deepslate replace minecraft:bedrock
+fill ~ 0 ~ ~15 0 ~15 minecraft:bedrock
```
[the remark this answers is not in the document]
Comment (misode) [reply gm4_orbis/data/gm4_orbis/functions/chunk/upgrade.mcfunction:4]: No, old chunks will only generate new terrain below Y=0 when they have bedrock at Y=0

File: gm4_orbis/data/gm4_orbis/functions/chunk/upgrade.mcfunction [hunk 2]
```diff
@@ -0,0 +1,4 @@
+fill ~ -63 ~ ~15 0 ~15 minecraft:deepslate
+clone ~ 1 ~ ~15 4 ~15 ~ -64 ~ filtered minecraft:bedrock
```
[the remark this answers is not in the document]
Comment (misode) [reply gm4_orbis/data/gm4_orbis/functions/chunk/upgrade.mcfunction]: Thanks for spotting this. Should be fixed now.

**Comment by BluePsychoRanger** [comment]
I wasn't sure about using markers as the detection system, but your system looks like it should work, since if one entity is loaded, all entities in that chunk is as well. With my 1.17 change that waited for the entity marker to be loaded in order to be initialized, all adjacent markers should be loaded when generating a chunk.

**Comment by misode** [comment]
Orbis 1.17 already used marker entities (although temporary), and would probably run into the same issues since the module attempts to spawn these markers outside the loaded chunks.
The logic now makes sure that before aligning and initializing a metachunk there is no existing initialized metachunk marker, otherwise it is killed. And it makes sure that there is a clock cycle between summoning the marker and attempting to target it.

File: gm4_tower_structures/data/gm4_tower_structures/functions/spawn/badlands.mcfunction [hunk 3]
```diff
@@ -1,3 +1,3 @@
-execute unless block ~ 4 ~ bedrock run summon item ~ ~ ~ {Tags:["gm4_orbis_structure"],Item:{id:"minecraft:stick",Count:1b,tag:{gm4_orbis:{name:"Tall Badlands Tower","structure":{"name":"gm4_tower_structures:badlands",posX:-3,posY:-1,posZ:-3}}}}}
-execute if block ~ 4 ~ bedrock unless block ~ 3 ~ bedrock run summon item ~ ~ ~ {Tags:["gm4_orbis_structure"],Item:{id:"minecraft:stick",Count:1b,tag:{gm4_orbis:{name:"Large Badlands Tower","structure":{"name":"gm4_tower_structures:badlands2",posX:-4,posY:-1,posZ:-4}}}}}
-execute if block ~ 4 ~ bedrock if block ~ 3 ~ bedrock run summon item ~ ~ ~ {Tags:["gm4_orbis_structure"],Item:{id:"minecraft:stick",Count:1b,tag:{gm4_orbis:{name:"Acacia Tower","structure":{"name":"gm4_tower_structures:badlands3",posX:-4,posY:-1,posZ:-4}}}}}
+execute unless block ~ -60 ~ bedrock run summon item ~ ~ ~ {Tags:["gm4_orbis_structure"],Item:{id:"minecraft:stick",Count:1b,tag:{gm4_orbis:{name:"Tall Badlands Tower","structure":{"name":"gm4_tower_structures:badlands",posX:-3,posY:-1,posZ:-3}}}}}
```
Comment (BluePsychoRanger) [inline gm4_tower_structures/data/gm4_tower_structures/functions/spawn/badlands.mcfunction:1]: Can/should this be moved to a loot table? (Apply this comment to the other `/spawn/<biome>` functions)

Comment (misode) [reply gm4_tower_structures/data/gm4_tower_structures/functions/spawn/badlands.mcfunction:1]: The idea here is that it's seed deterministic by checking the bedrock. But maybe that's not worth it and checking biomes, random chances and heights in a loot table would be a lot cleaner.
